backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/extraction/impl/LuceneCollectors.java

```diff
@@ -192,27 +197,20 @@ public TopDocs getTopDocs() {
 		return topDocs;
 	}
 
-	private void extractTopDocs(TopDocsCollector<?> topDocsCollector, int offset, Integer limit) {
-		if ( offset >= topDocsCollector.getTotalHits() ) {
+	private void extractTopDocs(int offset) {
+		if ( offset >= topDocs.totalHits.value ) {
 			// Hack.
 			// In this case, we cannot execute the code below as Lucene considers we passed incorrect arguments
 			// and returns a TopDocs instance with no ScoreDocs (correct)
 			// and a total hit count set to 0 (not always correct).
 			// Also, we cannot reconstruct the total hit count from just the collector
 			// since we don't have access to the relation (EQUAL/GT_OR_EQUAL).
 			// So we get just one topDoc, and infer everything from there.
-			TopDocs firstTopDoc = topDocsCollector.topDocs( 0, 1 );
-			topDocs = firstTopDoc instanceof TopFieldDocs
-					? new TopFieldDocs( firstTopDoc.totalHits, new FieldDoc[0], ( (TopFieldDocs) firstTopDoc ).fields )
-					: new TopDocs( firstTopDoc.totalHits, new ScoreDoc[0] );
-			return;
-		}
+			topDocs = topDocs instanceof TopFieldDocs
+					? new TopFieldDocs( topDocs.totalHits, new FieldDoc[0], ( (TopFieldDocs) topDocs ).fields )
+					: new TopDocs( topDocs.totalHits, new ScoreDoc[0] );
+
 
-		if ( limit == null ) {
-			topDocs = topDocsCollector.topDocs( offset );
-		}
-		else {
-			topDocs = topDocsCollector.topDocs( offset, limit );
 		}
 	}
```

Comment:
A few things:

1. You removed "the code below" mentioned in the comment, so the comment does not make sense anymore.
2. Will this even work, since you're no longer calling `.topDocs(0,1)` and the comment suggests that without that, we don't get a correct total hit count? I mean we should have integration tests for this and they passed, but it's probably worth double-checking (finding the relevant integration test).
3. Is this code even necesssary nowadays, and especially after your changes? Won't the topdocs already contain empty topdocs and a correct total hit count?

Reply:
> we have this one:

Wrong link? It doesn't look like the offset is higher than the document count or limit in this test.
Keep in mind that a `null` limit is essentially no limit, so higher than anything.

Reply:
Actually scratch that, the limit doesn't even matter. What we're after is a test where the offset is higher than the number of documents in the index.

Reply:
Why, you expected me to _scroll_? That's putting too much faith in my abilities.

Sorry :x  

Looks like the assertion you just quoted is missing something about the total hit count, but the assertion in `org.hibernate.search.integrationtest.backend.tck.search.query.SearchQueryFetchIT#fetch_offset_limit` and `org.hibernate.search.integrationtest.backend.tck.search.query.SearchQueryFetchIT#fetch_offset_limit_defaultSort` actually check the total hit count, so we're fine.

> I've dropped it and the Lucene tests still pass.

LGTM, thanks!